homeassistant/components/obihai/__init__.py

```diff
@@ -13,6 +15,33 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     return True
 
 
+async def async_migrate_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Migrate old entry."""
+
+    version = entry.version
+
+    LOGGER.debug("Migrating from version %s", version)
+    if version != 2:
+        requester = ObihaiConnection(
+            entry.data[CONF_HOST],
+            username=entry.data[CONF_USERNAME],
+            password=entry.data[CONF_PASSWORD],
+        )
+        await hass.async_add_executor_job(requester.update)
+
+        new_unique_id = await hass.async_add_executor_job(
+            requester.pyobihai.get_device_mac
+        )
+        if entry.unique_id != new_unique_id:
+            hass.config_entries.async_update_entry(entry, unique_id=new_unique_id)
```

Comment:
This doesn't need to be conditional, `async_update_entry` checks if the unique_id has been changed and if it hasn't returns without doing anything.
```suggestion
        hass.config_entries.async_update_entry(entry, unique_id=new_unique_id)
```